airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -47,13 +47,13 @@ class DatetimeBasedCursor(StreamSlicer):
     """
 
     start_datetime: Union[MinMaxDatetime, str]
-    end_datetime: Union[MinMaxDatetime, str]
     cursor_field: Union[InterpolatedString, str]
     datetime_format: str
     config: Config
     parameters: InitVar[Mapping[str, Any]]
     _cursor: dict = field(repr=False, default=None)  # tracks current datetime
     _cursor_end: dict = field(repr=False, default=None)  # tracks end of current stream slice
+    end_datetime: Union[MinMaxDatetime, str] = None
```

Comment:
nit should be Optional[Union[MinMaxDatetime, str]] now. and update the docstring too